omod/src/main/java/org/openmrs/module/oauth2login/web/Utils.java

```diff
@@ -56,7 +60,28 @@ public static String getPostLogoutRedirectUrl(HttpServletRequest request) throws
 			}
 		}
 		
+		if (!encodeDisabled) {
+			redirectPath = encodeUrl(redirectPath);
+		}
 		return StringUtils.defaultIfBlank(redirectPath, request.getContextPath() + "/oauth2login");
 	}
 	
+	protected static String encodeUrl(String logoutUrl) {
+		UriComponents urlComponents = UriComponentsBuilder.fromHttpUrl(logoutUrl).build();
+		MultiValueMap<String, String> params = urlComponents.getQueryParams();
+		UriComponentsBuilder urlBuilderEncoded = UriComponentsBuilder.newInstance();
+		urlBuilderEncoded.scheme(urlComponents.getScheme());
+		urlBuilderEncoded.host(urlComponents.getHost());
+		urlBuilderEncoded.port(urlComponents.getPort());
+		urlBuilderEncoded.path(urlComponents.getPath());
+		urlBuilderEncoded.fragment(urlComponents.getFragment());
+		if (params != null) {
+			for (Map.Entry<String, List<String>> entry : params.entrySet()) {
+				final String encodedKey = URLEncoder.encode(entry.getKey());
+				entry.getValue().forEach(v -> urlBuilderEncoded.queryParam(encodedKey, URLEncoder.encode(v)));
+			}
+		}
```

Comment:
> Empty query parameter is effectively a blank parameter value which is the same as any other parameter value in my opinion.

I get your point, I was suggesting to capture (or unit test) the above functionality. What is the expected outcome if `params == null`, can we capture that in tests

Reply:
I get your point, but is there an IDP that accepts a parameterless logout URL anyways? I intentionally omitted that test, in fact, for this reason I wanted to remove that if clause instead, params can never be null or empty otherwise that would be an invalid URL since the **client_id** or the **id_token_hint** parameter is required but this could be a Keycloak's implementation to require one of them, may be it is not the case for other implementations, so may be it make sense avoid a  potential NPE in case of other IDPs that might might accept a parameterless logout URL.

Reply:
➕ it's a utility method, someone might want to use it to encode a URL that's not specific to Keycloak's implementation.